### ISSUE TITLE
blake2-config.h `HAVE_SSE4_1` typo fix

### ISSUE THREAD

**Issue by nono303** [python/cpython#102027]
# Bug report

### Unable to compile cpython on Windows with **`AVX`** enable
1. adding `<EnableEnhancedInstructionSet>AdvancedVectorExtensions</EnableEnhancedInstructionSet>` for `</ClCompile>` in vcxproj files (result `/arch:AVX` in cl.exe cmd line)
2. `MSBuild.exe C:\sdk\src\python\PCbuild\pcbuild.proj /nologo /nr:false /m:16 /p:Turbo=true /p:CL_MPCount=16 /p:RunCodeAnalysis=false /p:DebugType=None /p:DebugSymbols=true /p:WindowsTargetPlatformVersion=10.0.22621.0 /p:PlatformToolset=v143 /clp:EnableMPLogging;Summary;ShowCommandLine /v:d /p:Configuration=Release /p:Platform="x64" /p:IncludeExtensions=true /p:IncludeExternals=true /p:IncludeTests=false /p:IncludeCTypes=true /p:IncludeSSL=true /p:IncludeTkinter=false /p:IncludeUwp=true /t:RebuildAll /p:EXTERNALS_DIR="C:\sdk\release\vs17_x64-avx" /p:ExternalsSrcDir="C:\sdk\src" /p:_DLLSuffix=-81_3-x64 /p:SqliteVersionStrToParse=3.40.1.0 /nowarn:C4244`
3. result:
> C:\sdk\src\python\Modules\_blake2\impl\blake2-config.h(67,1): fatal  error C1189: #error:  "This code requires at least SSE2." (compiling source file ..\Modules\_blake2\blake2s_impl.c) [C:\sdk\src\python\PCbuild\pythoncore.vcxproj]
> C:\sdk\src\python\Modules\_blake2\impl\blake2-config.h(67,1): fatal  error C1189: #error:  "This code requires at least SSE2." (compiling source file ..\Modules\_blake2\blake2b_impl.c) [C:\sdk\src\python\PCbuild\pythoncore.vcxproj]

### It's just a typo issue in `Modules/_blake2/impl/blake2-config.h`

```diff
diff --git "a/Modules/_blake2/impl/blake2-config.h" "b/Modules/_blake2/impl/blake2-config.h"
index f5dd6faa9e..c09cb4bcf0 100644
--- "a/Modules/_blake2/impl/blake2-config.h"
+++ "b/Modules/_blake2/impl/blake2-config.h"
@@ -53,7 +53,7 @@
 #endif
 #endif
 
-#ifdef HAVE_SSE41
+#ifdef HAVE_SSE4_1
 #ifndef HAVE_SSSE3
 #define HAVE_SSSE3
 #endif

```

The patch fix the issue
Seems to be present on every branch

# Your environment

- CPython versions tested on: branch 3.8 / tag 3.8.16
- Operating system and architecture: Win64 vs17 AVX

### Linked PRs
* gh-102124
* gh-102916
* gh-102917

**Comment by terryjreedy** [comment]
@tiran Can you verify that the proposed blake2 typo fix is correct?

**Comment by arhadthedev** [comment]
@nono303 We close bugfix issues after the corresponding PR is backported to previous versions, just to not lose track.

**Comment by YWtheGod** [comment]
But this fix hasn't merged into branch 3.9 and lower, people who want to build version before 3.10 still got this bug.

**Comment by terryjreedy** [comment]
This is standard. Non-security bugfixs are essentially never backported by us to versions that only get security-fixes.
